### Author : Rahul Bhoyar

Installing the necessary libraries

In [10]:
!pip install langchain langchain_openai faiss-cpu faiss-gpu docx pypdf2 reportlab PyPDF2

Setting up the memory object

In [3]:
from langchain.memory import ConversationBufferMemory
memory = ConversationBufferMemory(
    memory_key="chat_history",
    return_messages=True
)

Creating the retriever for initial response.

In [4]:
VECTOR_DATABASE_PATH = "vectorstore/db_faiss"

In [5]:
from langchain.vectorstores import FAISS
from langchain_openai import OpenAIEmbeddings

In [6]:
embeddings = OpenAIEmbeddings()

In [7]:
vectordb = FAISS.load_local(VECTOR_DATABASE_PATH, embeddings, allow_dangerous_deserialization = True)
print("Vector database loaded successfully.")
vectordb

Vector database loaded successfully.


In [8]:
# With specifying the number of arguments as  k = 15
retriever_for_intial_response = vectordb.as_retriever(search_kwargs={"k": 15})
print("Retriever created :",retriever_for_intial_response)

Retriever created : tags=['FAISS', 'OpenAIEmbeddings'] vectorstore=<langchain_community.vectorstores.faiss.FAISS object at 0x7ba53dc50550> search_kwargs={'k': 15}


Creating the get_kaggle_recommendations function.

In [11]:
from PyPDF2 import PdfReader
from langchain_openai import OpenAIEmbeddings
from langchain_community.vectorstores import FAISS
from langchain.memory import ConversationBufferMemory
from langchain.chains import ConversationalRetrievalChain
from langchain.prompts import PromptTemplate
from reportlab.lib.pagesizes import letter
from reportlab.platypus import SimpleDocTemplate,Paragraph
from reportlab.lib.styles import getSampleStyleSheet



def get_kaggle_recommendations(user_input,selected_action,llm,retriever):
    if selected_action == "Profile-Based Recommendation":
        custom_prompt_template = """
        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        {context}
        This is the extract of the text for which you need to find datasets:
        {question}
        At the end of the response, add the text: I hope this was a helpful response. Now you can talk with the recommended data.
        """

    elif selected_action == "Expert-Based Recommendation":
        custom_prompt_template = """
        Based on the Project Exposé, KaggleGPT combines current trends and topics in the fields and proposes challenging ideas with datasets. The output here is intended for good students who want to pursue challenging ideas.
        Your tasks are as follows:
        1. You should summarize several current interesting trends to persuade students working on challenging datasets.
        2. You should display results in a table for easy viewing.
        3. You should provide at least 8 different datasets.
        4. You should sort the datasets by size and usabilityRating. The larger the size and usabilityRating, the more difficult to work with those datasets.
        5. You should provide extra advances, such as requiring students to consider using powerful computing systems or cloud platforms to work with large datasets, developing a runnable prototype, or deploying a demo.

        However, if you see that the project proposal has mentioned and given information that are aligned with most of your provided tasks. You can simply respond
by saying, “The Project Exposé is in good shape. Please book a consultation with your supervisor.”

        Below is the context :
        {context}
        This is the extract of the text for which you need to find datasets:
        {question}
        At the end of the response, add the text: I hope this was a helpful response. Now you can talk with the recommended data.
        """

    elif selected_action == "Knowledge-Based Recommendation":
        custom_prompt_template = """
        The outputs are purely based on the master programs and syllabus with fixed learning outcomes. How a regular project should look like. Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.
        4. You should sort the datasets by viewCount and voteCount. The larger the viewCount and voteCount, the more popular to work with those datasets.

        However, if you see that the project proposal has mentioned and given information that is aligned with most of your provided tasks. You can simply respond
by saying, “The Project Exposé is in good shape. Please book a consultation with your supervisor.”

        Below is the context :
        {context}
        This is the extract of the text for which you need to find datasets:
        {question}
        At the end of the response, add the text: I hope this was a helpful response. Now you can talk with the recommended data.
        """

    elif selected_action == "Multi-Criteria Based Recommendation":
        custom_prompt_template = """
        The combined recommendation considers other meta information, such as how long is the thesis duration. Is the topic suitable for the restricted time frame? Do students invest in GPU workstation or cloud computing to run experiments? Do students want to have a conference and journal submission out of the results. KaggleGPT might ask the students if they have the required criteria. Your tasks are as follows:
        1. You should summarize several current interesting trends to persuade students working on challenging datasets.
        2. You should display results in a table for easy viewing.
        3. You should provide at least 8 different datasets.
        4. You should sort the datasets by size, usabilityRating, viewCount and voteCount. The larger the numbers, the more challenging to work with those datasets.
        5. You should mention that submitting a research paper is highly recommended.
        6. You should provide extra advances, such as requiring students to consider using powerful computing systems or cloud platforms to work with big datasets. Students must also develop a runnable prototype or deploy a demo.

        However, if you see that the project proposal mentions and provides information that is aligned with most of your provided tasks, you can simply respond by saying, “The Project Exposé is in good shape. Please book a consultation with your supervisor.”

        Below is the context :
        {context}
        This is the extract of the text for which you need to find datasets:
        {question}
        At the end of the response, add the text: I hope this was a helpful response. Now you can talk with the recommended data.
        """

    # Create a PromptTemplate instance with your custom template
    custom_prompt = PromptTemplate(
        template=custom_prompt_template,
        input_variables=["context", "question"],
    )
    # Use your custom prompt when creating the ConversationalRetrievalChain

    memory = ConversationBufferMemory(
    memory_key="chat_history",
    return_messages=True
     )

    qa = ConversationalRetrievalChain.from_llm(
        llm,
        verbose=True,
        retriever=retriever,
        memory=memory,
        combine_docs_chain_kwargs={"prompt": custom_prompt},
    )
    response = qa({"question": user_input})['answer']
    return response

In [15]:
SAMPLE_TEXT = """
I hope this message finds you well. My name is Narasimha Krishna, and I am currently
pursuing my second master's degree in Artificial Intelligence, with a keen focus on Data
Science and Engineering. Your esteemed background as a Senior Researcher in AI has
inspired me greatly, and I am reaching out to express my sincere interest in the possibility
of conducting my Master's thesis under your supervision.
I am deeply passionate about the field of Data Science and Engineering, and I am
committed to contributing innovative research that pushes the boundaries of knowledge
in this domain. Given your expertise and experience, I believe that collaborating with you
would provide me with invaluable guidance and mentorship to refine my skills and tackle
challenging problems in the field.
I am eager to embark on a thesis project that not only aligns with your areas of interest but
also offers a unique contribution to the field. I am confident in my ability to work diligently
and persistently, even under pressure, as I am someone who thrives on challenges and
never shies away from hard work.
I understand that you may have numerous demands on your time, but I would be
immensely grateful for the opportunity to discuss potential thesis topics and explore how
my research interests could complement your ongoing work. Your mentorship would be
instrumental in shaping my academic and professional trajectory, and I am fully
committed to making the most of this opportunity.
Thank you for considering my request. I eagerly await your positive response and the
possibility of working together on an exciting and impactful research project.

"""

### (A) OpenAI LLM Model Response

Initialising the model.

In [16]:
import os
openai_api_key = "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"  # Enter your OPENAI_API_KEY
os.environ["OPENAI_API_KEY"] = openai_api_key
print("OPENAI API key is set successfully :",openai_api_key)

OPENAI API key is set successfully : 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


In [17]:
from langchain_openai import ChatOpenAI
openai_llm = ChatOpenAI(model_name = "gpt-4", streaming=True)
openai_llm

ChatOpenAI(client=<openai.resources.chat.completions.Completions object at 0x7ba50575ac50>, async_client=<openai.resources.chat.completions.AsyncCompletions object at 0x7ba50456aa10>, model_name='gpt-4', openai_api_key=SecretStr('**********'), openai_proxy='', streaming=True)

In [18]:
selected_action = "Profile-Based Recommendation"
response_from_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,openai_llm,retriever_for_intial_response)
response_from_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

"Here are the recommended datasets based on the context provided:\n\n| Dataset Number | Dataset Name | Description | Topic/Tags |\n| -------------- | ------------ | ----------- | ---------- |\n| 1479 | Job Dataset | A Comprehensive Job Dataset for Data Science, Research, and Analysis | Computer Science, Data Analytics |\n| 980 | AI Robotics Employee Salary | A Comprehensive Resource for Predicting Salaries and Understanding Salary Trends | Data Visualization, Data Cleaning |\n| 1586 | Data Science Interview Questions | Questions asked by - FAANG/Fortune 500/Top Product Companies in last 4-5 years | Computer Science, NLP, Deep Learning |\n| 1690 | Resume Dataset | A collection of Resumes in PDF as well as String format for data extraction | NLP, Text Mining |\n| 2047 | Wipro's Sustainability Machine Learning Challenge | Wipro's Sustainability Machine Learning Challenge | Machine Learning, Sustainability |\n| 1634 | Microsoft Professional Capstone DataSet | EdX Microsoft Capstone Project

Response :

Here are the recommended datasets based on the context provided:

| Dataset Number | Dataset Name | Description | Topic/Tags |
| -------------- | ------------ | ----------- | ---------- |
| 1479 | Job Dataset | A Comprehensive Job Dataset for Data Science, Research, and Analysis | Computer Science, Data Analytics |
| 980 | AI Robotics Employee Salary | A Comprehensive Resource for Predicting Salaries and Understanding Salary Trends | Data Visualization, Data Cleaning |
| 1586 | Data Science Interview Questions | Questions asked by - FAANG/Fortune 500/Top Product Companies in last 4-5 years | Computer Science, NLP, Deep Learning |
| 1690 | Resume Dataset | A collection of Resumes in PDF as well as String format for data extraction | NLP, Text Mining |
| 2047 | Wipro's Sustainability Machine Learning Challenge | Wipro's Sustainability Machine Learning Challenge | Machine Learning, Sustainability |
| 1634 | Microsoft Professional Capstone DataSet | EdX Microsoft Capstone Project DataSet | Education, Finance, Demographics |
| 2044 | NLP on Research Articles | Multi Label Classification using NLP on Research Articles | NLP, Multi Label Classification |
| 1013 | Glassdoor Data Science Jobs - 2024 | 900 Real data science job listings from Glassdoor India - Dec'23 - Jan'24 | Data Visualization, NLP |
| 1761 | IRIT Researchers Database | Dataset of the latest research contribution by members of IRIT, Toulouse France | AI, Programming, Engineering |
| 1478 | Employee Dataset | Employee Dataset ( Training, Survey, Performance, Recruitment, Attendance) | Data Analytics, HR |
| 1645 | HR Analytics Analytics Vidya | HR analytics is revolutionising the way human resources departments operate | Business, Software |
| 1425 | DeepMind Research Papers | Research Papers on AI, Neuroscience, etc published by DeepMind, Google | AI, Neuroscience, Deep Learning |

I hope this was a helpful response. Now you can talk with the recommended data.

### (B) facebook/bart-large-cnn Model Response

Initialisig the model.

In [20]:
hf_api_key = "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"
os.environ["HUGGINGFACEHUB_API_TOKEN"] = hf_api_key
print("HUGGINGFACEHUB_API_TOKEN is set successfully :",hf_api_key)


HUGGINGFACEHUB_API_TOKEN is set successfully : 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


In [34]:
from langchain import HuggingFaceHub

fb_bart_large_llm = HuggingFaceHub(repo_id="facebook/bart-large-cnn")


fb_bart_large_llm

HuggingFaceHub(client=<InferenceClient(model='facebook/bart-large-cnn', timeout=None)>, repo_id='facebook/bart-large-cnn', task='summarization')

In [35]:
selected_action = "Profile-Based Recommendation"
response_from_fb_bart_large_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,fb_bart_large_llm,retriever_for_intial_response)
response_from_fb_bart_large_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

'Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.'

Response:

Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.

### (C) Google Gemma 7B Model Response

In [32]:
from langchain import HuggingFaceHub

gemma_llm = HuggingFaceHub(repo_id="google/gemma-7b")
gemma_llm

HuggingFaceHub(client=<InferenceClient(model='google/gemma-7b', timeout=None)>, repo_id='google/gemma-7b', task='text-generation')

In [33]:
selected_action = "Profile-Based Recommendation"
response_from_gemma_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,gemma_llm,retriever_for_intial_response)
response_from_gemma_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

"\n        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.\n        Your tasks are as follows:\n        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.\n        2. You should display results in a table for easy viewing.\n        3. You should group datasets by topic.\n\n        Below is the context :\n        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/ravindrasinghrana/job-description-dataset d

Response :



Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

Below is the context :

dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/ravindrasinghrana/job-description-dataset description_tokens: 59 tags: [universities and colleges, computer science, exploratory data analysis, data analytics, jobs and career]

dataset-Nr.: 980 dataset_name: kiranpatil7022/ai-robotics-employee-salary title: ai robotics employee salary description: ai robotics employee salary by Kiran Patil. A Comprehensive Resource for Predicting Salaries and Understanding Salary Trends Last updated on 26/10/2023 17:05. Size: 23MB. Usability rating: 0.76. View count: 271, Vote count: 1. Available under Apache 2.0 license. Find more at: https://www.kaggle.com/datasets/kiranpatil7022/ai-robotics-employee-salary description_tokens: 56 tags: [exploratory data analysis, data cleaning, data visualization, data analytics, regression]

dataset-Nr.: 192 dataset_name: anthonytherrien/academic-prompts-collection title: Academic Inquiry Unleashed description: Academic Inquiry Unleashed by AnthonyTherrien. Navigating the Realms of Knowledge with teknium/OpenHermes-2p5-Mistral-7B's Last updated on 01/02/2024 23:18. Size: 5MB. Usability rating: 1.00. View count: 222, Vote count: 5. Available under CC BY-SA 4.0 license. Find more at: https://www.kaggle.com/datasets/anthonytherrien/academic-prompts-collection description_tokens: 53 tags: [universities and colleges, earth and nature, education, nlp, text]

dataset-Nr.: 1586 dataset_name: sandy1811/data-science-interview-questions title: Data Science Interview Questions description: Data Science Interview Questions by Sandeep Singh. Questions asked by - FAANG/Fortune 500/Top Product Companies in last 4-5 years Last updated on 11/02/2024 05:27. Size: 1MB. Usability rating: 0.94. View count: 14942, Vote count: 74. Available under Data files © Original Authors license. Find more at: https://www.kaggle.com/datasets/sandy1811/data-science-interview-questions description_tokens: 61 tags: [employment, computer science, nlp, computer vision, deep learning]

dataset-Nr.: 1690 dataset_name: snehaanbhawal/resume-dataset title: Resume Dataset description: Resume Dataset by Snehaan Bhawal. A collection of Resumes in PDF as well as String format for data extraction. Last updated on 08/08/2021 10:52. Size: 62MB. Usability rating: 1.00. View count: 79680, Vote count: 126. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/snehaanbhawal/resume-dataset description_tokens: 61 tags: [business, nlp, text mining, text, spaCy]

dataset-Nr.: 2047 dataset_name: vickeytomer/wipros-sustainability-machine-learning-challenge title: Wipro's Sustainability Machine Learning Challenge description: Wipro's Sustainability Machine Learning Challenge by vivek kumar. Wipro's Sustainability Machine Learning Challenge Last updated on 14/03/2022 10:56. Size: 3MB. Usability rating: 0.71. View count: 5880, Vote count: 27. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/vickeytomer/wipros-sustainability-machine-learning-challenge description_tokens: 56 tags: [earth and nature, education, energy]

dataset-Nr.: 1634 dataset_name: sharmaharsh/microsoft-capstone title: Microsoft Professional Capstone DataSet description: Microsoft Professional Capstone DataSet by Harsh Sharma. EdX Microsoft Capstone Project DataSet Last updated on 22/10/2017 02:14. Size: 7MB. Usability rating: 0.53. View count: 7614, Vote count: 9. Available under Unknown license. Find more at: https://www.kaggle.com/datasets/sharmaharsh/microsoft-capstone description_tokens: 50 tags: [universities and colleges, business, education, finance, demographics, software]

dataset-Nr.: 2044 dataset_name: vetrirah/janatahack-independence-day-2020-ml-hackathon title: NLP on Research Articles description: NLP on Research Articles by Vetrivel-PS. Multi Label Classification using NLP on Research Articles Last updated on 19/08/2020 14:35. Size: 11MB. Usability rating: 1.00. View count: 14495, Vote count: 39. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/vetrirah/janatahack-independence-day-2020-ml-hackathon description_tokens: 55 tags: [earth and nature, business, science and technology, computer science, tpu, gpu]

dataset-Nr.: 36 dataset_name: abisheksudarshan/topic-modeling-for-research-articles title: Topic Modeling for Research Articles description: Topic Modeling for Research Articles by Abishek Sudarshan. NLP Dataset for Beginners Last updated on 01/01/2022 17:17. Size: 7MB. Usability rating: 1.00. View count: 14313, Vote count: 24. Available under Data files © Original Authors license. Find more at: https://www.kaggle.com/datasets/abisheksudarshan/topic-modeling-for-research-articles description_tokens: 54 tags: [earth and nature, education, psychology, nlp]

dataset-Nr.: 1013 dataset_name: kuralamuthan300/glassdoor-data-science-jobs title: Glassdoor Data science Jobs - 2024 description: Glassdoor Data science Jobs - 2024 by Kuralamuthan Kathirvelan. 900 Real data science job listings from Glassdoor India - Dec'23 - Jan'24 Last updated on 16/01/2024 04:09. Size: 800KB. Usability rating: 1.00. View count: 5765, Vote count: 28. Available under MIT license. Find more at: https://www.kaggle.com/datasets/kuralamuthan300/glassdoor-data-science-jobs description_tokens: 60 tags: [computer science, intermediate, exploratory data analysis, nlp, data visualization, tabular, jobs and career]

dataset-Nr.: 1761 dataset_name: syedhaseebahmad/irit-researchers-database title: IRIT Researchers database description: IRIT Researchers database by syed haseeb ahmad. Dataset of the latest research contribution by members of IRIT, Toulouse France Last updated on 27/11/2022 13:12. Size: 276KB. Usability rating: 0.53. View count: 541, Vote count: 4. Available under Unknown license. Find more at: https://www.kaggle.com/datasets/syedhaseebahmad/irit-researchers-database description_tokens: 58 tags: [research, employment, business, artificial intelligence, programming, engineering]

dataset-Nr.: 1478 dataset_name: ravindrasinghrana/employeedataset title: Employee Dataset(All in One) description: Employee Dataset(All in One) by Ravender Singh Rana. Employee Dataset ( Training, Survey, Performance, Recruitment, Attendance) Last updated on 13/08/2023 08:06. Size: 520KB. Usability rating: 1.00. View count: 21976, Vote count: 68. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/ravindrasinghrana/employeedataset description_tokens: 65 tags: [universities and colleges, employment, intermediate, advanced, data analytics]

dataset-Nr.: 584 dataset_name: emirhanai/artificial-intelligence-project-harvard-university title: Artificial Intelligence Project Harvard University description: Artificial Intelligence Project Harvard University by EMnRHAN BULUT. My Artificial Intelligence Project (Tic Tac Toe) - Harvard University Last updated on 12/04/2023 11:13. Size: 578KB. Usability rating: 1.00. View count: 4946, Vote count: 39. Available under Attribution 4.0 International (CC BY 4.0) license. Find more at: https://www.kaggle.com/datasets/emirhanai/artificial-intelligence-project-harvard-university description_tokens: 65 tags: [universities and colleges, education, artificial intelligence, computer

dataset-Nr.: 1645 dataset_name: shikhnu/hr-analytics-analytics-vidya title: HR Analytics Analytics Vidya description: HR Analytics Analytics Vidya by Mohammad Imran Shaikh. HR analytics is revolutionising the way human resources departments operate. Last updated on 08/10/2020 16:44. Size: 938KB. Usability rating: 0.76. View count: 7828, Vote count: 9. Available under Unknown license. Find more at: https://www.kaggle.com/datasets/shikhnu/hr-analytics-analytics-vidya description_tokens: 57 tags: [employment, business, software]

dataset-Nr.: 1425 dataset_name: ppb00x/deepmind-research-papers title: DeepMind Research Papers description: DeepMind Research Papers by n Srihari. Research Papers on AI, Neuroscience, etc published by DeepMind, Google. Last updated on 27/08/2023 11:01. Size: 2GB. Usability rating: 0.88. View count: 336, Vote count: 4. Available under EU ODP Legal Notice license. Find more at: https://www.kaggle.com/datasets/ppb00x/deepmind-research-papers description_tokens: 61 tags: [neuroscience, artificial intelligence, computer science, deep learning, neural networks]
        This is the extract of the text for which you need to find datasets:
        
I hope this message finds you well. My name is Narasimha Krishna, and I am currently
pursuing my second master's degree in Artificial Intelligence, with a keen focus on Data
Science and Engineering. Your esteemed background as a Senior Researcher in AI has
inspired me greatly, and I am reaching out to express my sincere interest in the possibility
of conducting my Master's thesis under your supervision.
I am deeply passionate about the field of Data Science and Engineering, and I am
committed to contributing innovative research that pushes the boundaries of knowledge
in this domain. Given your expertise and experience, I believe that collaborating with you
would provide me with invaluable guidance and mentorship to refine my skills and tackle
challenging problems in the field.
I am eager to embark on a thesis project that not only aligns with your areas of interest but
also offers a unique contribution to the field. I am confident in my ability to work diligently
and persistently, even under pressure, as I am someone who thrives on challenges and
never shies away from hard work.
I understand that you may have numerous demands on your time, but I would be
immensely grateful for the opportunity to discuss potential thesis topics and explore how
my research interests could complement your ongoing work. Your mentorship would be
instrumental in shaping my academic and professional trajectory, and I am fully
committed to making the most of this opportunity.
Thank you for considering my request. I eagerly await your positive response and the
possibility of working together on an exciting and impactful research project.

At the end of the response, add the text: I hope this was a helpful response. Now you can talk with the recommended data.
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        
        

### (C) meta-llama/Llama-2-70b-hf Model Response

In [36]:
from langchain import HuggingFaceHub

meta_llama_llm = HuggingFaceHub(repo_id="meta-llama/Llama-2-70b-hf")
meta_llama_llm

HuggingFaceHub(client=<InferenceClient(model='meta-llama/Llama-2-70b-hf', timeout=None)>, repo_id='meta-llama/Llama-2-70b-hf', task='text-generation')

In [44]:
selected_action = "Profile-Based Recommendation"
response_from_meta_llama_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,meta_llama_llm,retriever_for_intial_response)
response_from_meta_llama_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

HfHubHTTPError: 403 Client Error: Forbidden for url: https://api-inference.huggingface.co/models/meta-llama/Llama-2-70b-hf (Request ID: yzrVGV33uE6PsL4XehQVs)

The model meta-llama/Llama-2-70b-hf is too large to be loaded automatically (137GB > 10GB). Please use Spaces (https://huggingface.co/spaces) or Inference Endpoints (https://huggingface.co/inference-endpoints).

### (D) mistralai/Mixtral-8x7B-Instruct-v0.1 model response

In [39]:
from langchain import HuggingFaceHub

mixtral_llm = HuggingFaceHub(repo_id="mistralai/Mixtral-8x7B-Instruct-v0.1")
mixtral_llm

HuggingFaceHub(client=<InferenceClient(model='mistralai/Mixtral-8x7B-Instruct-v0.1', timeout=None)>, repo_id='mistralai/Mixtral-8x7B-Instruct-v0.1', task='text-generation')

In [40]:
selected_action = "Profile-Based Recommendation"
response_from_mixtral_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,mixtral_llm,retriever_for_intial_response)
response_from_mixtral_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

"\n        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.\n        Your tasks are as follows:\n        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.\n        2. You should display results in a table for easy viewing.\n        3. You should group datasets by topic.\n\n        Below is the context :\n        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/ravindrasinghrana/job-description-dataset d

Response :


Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
  
  Your tasks are as follows:
  1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.

  2. You should display results in a table for easy viewing.
  
  3. You should group datasets by topic.

Below is the context :

dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/ravindrasinghrana/job-description-dataset description_tokens: 59 tags: [universities and colleges, computer science, exploratory data analysis, data analytics, jobs and career]

dataset-Nr.: 980 dataset_name: kiranpatil7022/ai-robotics-employee-salary title: ai robotics employee salary description: ai robotics employee salary by Kiran Patil. A Comprehensive Resource for Predicting Salaries and Understanding Salary Trends Last updated on 26/10/2023 17:05. Size: 23MB. Usability rating: 0.76. View count: 271, Vote count: 1. Available under Apache 2.0 license. Find more at: https://www.kaggle.com/datasets/kiranpatil7022/ai-robotics-employee-salary description_tokens: 56 tags: [exploratory data analysis, data cleaning, data visualization, data analytics, regression]

dataset-Nr.: 192 dataset_name: anthonytherrien/academic-prompts-collection title: Academic Inquiry Unleashed description: Academic Inquiry Unleashed by AnthonyTherrien. Navigating the Realms of Knowledge with teknium/OpenHermes-2p5-Mistral-7B's Last updated on 01/02/2024 23:18. Size: 5MB. Usability rating: 1.00. View count: 222, Vote count: 5. Available under CC BY-SA 4.0 license. Find more at: https://www.kaggle.com/datasets/anthonytherrien/academic-prompts-collection description_tokens: 53 tags: [universities and colleges, earth and nature, education, nlp, text]

dataset-Nr.: 1586 dataset_name: sandy1811/data-science-interview-questions title: Data Science Interview Questions description: Data Science Interview Questions by Sandeep Singh. Questions asked by - FAANG/Fortune 500/Top Product Companies in last 4-5 years Last updated on 11/02/2024 05:27. Size: 1MB. Usability rating: 0.94. View count: 14942, Vote count: 74. Available under Data files © Original Authors license. Find more at: https://www.kaggle.com/datasets/sandy1811/data-science-interview-questions description_tokens: 61 tags: [employment, computer science, nlp, computer vision, deep learning]

dataset-Nr.: 1690 dataset_name: snehaanbhawal/resume-dataset title: Resume Dataset description: Resume Dataset by Snehaan Bhawal. A collection of Resumes in PDF as well as String format for data extraction. Last updated on 08/08/2021 10:52. Size: 62MB. Usability rating: 1.00. View count: 79680, Vote count: 126. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/snehaanbhawal/resume-dataset description_tokens: 61 tags: [business, nlp, text mining, text, spaCy]

dataset-Nr.: 2047 dataset_name: vickeytomer/wipros-sustainability-machine-learning-challenge title: Wipro's Sustainability Machine Learning Challenge description: Wipro's Sustainability Machine Learning Challenge by vivek kumar. Wipro's Sustainability Machine Learning Challenge Last updated on 14/03/2022 10:56. Size: 3MB. Usability rating: 0.71. View count: 5880, Vote count: 27. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/vickeytomer/wipros-sustainability-machine-learning-challenge description_tokens: 56 tags: [earth and nature, education, energy]

dataset-Nr.: 1634 dataset_name: sharmaharsh/microsoft-capstone title: Microsoft Professional Capstone DataSet description: Microsoft Professional Capstone DataSet by Harsh Sharma. EdX Microsoft Capstone Project DataSet Last updated on 22/10/2017 02:14. Size: 7MB. Usability rating: 0.53. View count: 7614, Vote count: 9. Available under Unknown license. Find more at: https://www.kaggle.com/datasets/sharmaharsh/microsoft-capstone description_tokens: 50 tags: [universities and colleges, business, education, finance, demographics, software]

dataset-Nr.: 2044 dataset_name: vetrirah/janatahack-independence-day-2020-ml-hackathon title: NLP on Research Articles description: NLP on Research Articles by Vetrivel-PS. Multi Label Classification using NLP on Research Articles Last updated on 19/08/2020 14:35. Size: 11MB. Usability rating: 1.00. View count: 14495, Vote count: 39. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/vetrirah/janatahack-independence-day-2020-ml-hackathon description_tokens: 55 tags: [earth and nature, business, science and technology, computer science, tpu, gpu]

dataset-Nr.: 36 dataset_name: abisheksudarshan/topic-modeling-for-research-articles title: Topic Modeling for Research Articles description: Topic Modeling for Research Articles by Abishek Sudarshan. NLP Dataset for Beginners Last updated on 01/01/2022 17:17. Size: 7MB. Usability rating: 1.00. View count: 14313, Vote count: 24. Available under Data files © Original Authors license. Find more at: https://www.kaggle.com/datasets/abisheksudarshan/topic-modeling-for-research-articles description_tokens: 54 tags: [earth and nature, education, psychology, nlp]

dataset-Nr.: 1013 dataset_name: kuralamuthan300/glassdoor-data-science-jobs title: Glassdoor Data science Jobs - 2024 description: Glassdoor Data science Jobs - 2024 by Kuralamuthan Kathirvelan. 900 Real data science job listings from Glassdoor India - Dec'23 - Jan'24 Last updated on 16/01/2024 04:09. Size: 800KB. Usability rating: 1.00. View count: 5765, Vote count: 28. Available under MIT license. Find more at: https://www.kaggle.com/datasets/kuralamuthan300/glassdoor-data-science-jobs description_tokens: 60 tags: [computer science, intermediate, exploratory data analysis, nlp, data visualization, tabular, jobs and career]

dataset-Nr.: 1761 dataset_name: syedhaseebahmad/irit-researchers-database title: IRIT Researchers database description: IRIT Researchers database by syed haseeb ahmad. Dataset of the latest research contribution by members of IRIT, Toulouse France Last updated on 27/11/2022 13:12. Size: 276KB. Usability rating: 0.53. View count: 541, Vote count: 4. Available under Unknown license. Find more at: https://www.kaggle.com/datasets/syedhaseebahmad/irit-researchers-database description_tokens: 58 tags: [research, employment, business, artificial intelligence, programming, engineering]

dataset-Nr.: 1478 dataset_name: ravindrasinghrana/employeedataset title: Employee Dataset(All in One) description: Employee Dataset(All in One) by Ravender Singh Rana. Employee Dataset ( Training, Survey, Performance, Recruitment, Attendance) Last updated on 13/08/2023 08:06. Size: 520KB. Usability rating: 1.00. View count: 21976, Vote count: 68. Available under CC0: Public Domain license. Find more at: https://www.kaggle.com/datasets/ravindrasinghrana/employeedataset description_tokens: 65 tags: [universities and colleges, employment, intermediate, advanced, data analytics]

dataset-Nr.: 584 dataset_name: emirhanai/artificial-intelligence-project-harvard-university title: Artificial Intelligence Project Harvard University description: Artificial Intelligence Project Harvard University by EMnRHAN BULUT. My Artificial Intelligence Project (Tic Tac Toe) - Harvard University Last updated on 12/04/2023 11:13. Size: 578KB. Usability rating: 1.00. View count: 4946, Vote count: 39. Available under Attribution 4.0 International (CC BY 4.0) license. Find more at: https://www.kaggle.com/datasets/emirhanai/artificial-intelligence-project-harvard-university description_tokens: 65 tags: [universities and colleges, education, artificial intelligence, computer

dataset-Nr.: 1645 dataset_name: shikhnu/hr-analytics-analytics-vidya title: HR Analytics Analytics Vidya description: HR Analytics Analytics Vidya by Mohammad Imran Shaikh. HR analytics is revolutionising the way human resources departments operate. Last updated on 08/10/2020 16:44. Size: 938KB. Usability rating: 0.76. View count: 7828, Vote count: 9. Available under Unknown license. Find more at: https://www.kaggle.com/datasets/shikhnu/hr-analytics-analytics-vidya description_tokens: 57 tags: [employment, business, software]

dataset-Nr.: 1425 dataset_name: ppb00x/deepmind-research-papers title: DeepMind Research Papers description: DeepMind Research Papers by n Srihari. Research Papers on AI, Neuroscience, etc published by DeepMind, Google. Last updated on 27/08/2023 11:01. Size: 2GB. Usability rating: 0.88. View count: 336, Vote count: 4. Available under EU ODP Legal Notice license. Find more at: https://www.kaggle.com/datasets/ppb00x/deepmind-research-papers description_tokens: 61 tags: [neuroscience, artificial intelligence, computer science, deep learning, neural networks]
        This is the extract of the text for which you need to find datasets:
        
I hope this message finds you well. My name is Narasimha Krishna, and I am currently
pursuing my second master's degree in Artificial Intelligence, with a keen focus on Data
Science and Engineering. Your esteemed background as a Senior Researcher in AI has
inspired me greatly, and I am reaching out to express my sincere interest in the possibility
of conducting my Master's thesis under your supervision.
I am deeply passionate about the field of Data Science and Engineering, and I am
committed to contributing innovative research that pushes the boundaries of knowledge
in this domain. Given your expertise and experience, I believe that collaborating with you
would provide me with invaluable guidance and mentorship to refine my skills and tackle
challenging problems in the field.
I am eager to embark on a thesis project that not only aligns with your areas of interest but
also offers a unique contribution to the field. I am confident in my ability to work diligently
and persistently, even under pressure, as I am someone who thrives on challenges and
never shies away from hard work.
I understand that you may have numerous demands on your time, but I would be
immensely grateful for the opportunity to discuss potential thesis topics and explore how
my research interests could complement your ongoing work. Your mentorship would be
instrumental in shaping my academic and professional trajectory, and I am fully
committed to making the most of this opportunity.
Thank you for considering my request. I eagerly await your positive response and the
possibility of working together on an exciting and impactful research project.


At the end of the response, add the text: I hope this was a helpful response. Now you can talk with the recommended data.
        
Here is the response:

Dear Narasimha Krishna,

Thank you for your interest in conducting your Master's thesis under my supervision. I am honored to have the opportunity to guide you in your research journey.

After reviewing your background and interests, I have compiled a list of 10 datasets that align with your focus on Data Science and Engineering. These datasets cover various topics in computer vision, natural language processing,

### (F) Zephyr-7b-beta Model Response

In [41]:
from langchain_community.llms import HuggingFaceEndpoint
from langchain_community.chat_models.huggingface import ChatHuggingFace

zephyr_llm = HuggingFaceEndpoint(repo_id="HuggingFaceH4/zephyr-7b-beta")
zephyr_llm

Token will not been saved to git credential helper. Pass `add_to_git_credential=True` if you want to set the git credential as well.
Token is valid (permission: write).
Your token has been saved to /root/.cache/huggingface/token
Login successful


In [42]:
selected_action = "Profile-Based Recommendation"
response_from_zephyr_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,zephyr_llm,retriever_for_intial_response)
response_from_zephyr_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

'\n        And the person can reply: Thanks for your help. Can you please provide me with some more datasets related to time series analysis? I am particularly interested in financial time series data.'

Response:


  And the person can reply: Thanks for your help. Can you please provide me with some more datasets related to time series analysis? I am particularly interested in financial time series data.

### (G) microsoft/phi-2

In [55]:
from langchain_community.llms import HuggingFaceEndpoint
from langchain_community.chat_models.huggingface import ChatHuggingFace

phi2_llm = HuggingFaceEndpoint(repo_id="microsoft/phi-2",timeout=500)
phi2_llm

Token will not been saved to git credential helper. Pass `add_to_git_credential=True` if you want to set the git credential as well.
Token is valid (permission: write).
Your token has been saved to /root/.cache/huggingface/token
Login successful


HuggingFaceEndpoint(repo_id='microsoft/phi-2', timeout=500, model='microsoft/phi-2', client=<InferenceClient(model='microsoft/phi-2', timeout=500)>, async_client=<InferenceClient(model='microsoft/phi-2', timeout=500)>)

In [56]:
selected_action = "Profile-Based Recommendation"
response_phi2_llm = get_kaggle_recommendations(SAMPLE_TEXT,selected_action,phi2_llm,retriever_for_intial_response)
response_phi2_llm



> Entering new StuffDocumentsChain chain...


> Entering new LLMChain chain...
Prompt after formatting:

        Project Exposé must be in computer science, machine learning, and artificial intelligence in general. Students might not give a precise description and ideas. They just want to have any proposed topics with datasets.
        Your tasks are as follows:
        1. You should provide at least 10 different datasets on computer vision, natural language processing, or time series.
        2. You should display results in a table for easy viewing.
        3. You should group datasets by topic.

        Below is the context :
        dataset-Nr.: 1479 dataset_name: ravindrasinghrana/job-description-dataset title: Job Dataset description: Job Dataset by Ravender Singh Rana. A Comprehensive Job Dataset for Data Science, Research, and Analysis Last updated on 17/09/2023 15:52. Size: 457MB. Usability rating: 1.00. View count: 23711, Vote count: 78. Available under CC0: Public Domain l

HfHubHTTPError: 429 Client Error: Too Many Requests for url: https://api-inference.huggingface.co/models/microsoft/phi-2 (Request ID: fY8zjkugn8kbLn9pJxmQx)

Rate limit reached. You reached free usage limit (reset hourly). Please subscribe to a plan at https://huggingface.co/pricing to use the API at this rate